packages/utils/src/types.ts

```diff
@@ -688,18 +690,13 @@ export type ArrayFieldItemTemplateType<
   /** A boolean value stating if the array item is read-only */
   readonly?: boolean;
   /** A stable, unique key for the array item */
-  key: string;
+  itemKey: string;
+  /** The UI schema of the array item's parent array field used for
+   * customization in some themes
+   */
+  parentUiSchema?: UiSchema<T, S, F>;
 };
 
-/**
```

Comment:
Since we've made such a large set of breaking changes, I figured we don't need to maintain this anymore